### PR TITLE
Feat(eos_designs): Add filter.allow_vrfs and filter.deny_vrfs

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Add filter.vrfs and filter.not_vrfs

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

```
          # Filter L3 and L2 network services based on tenant and tags (and operation filter).
          # If filter is not defined it will default to all.
          filter:

            # Limit configured Network Services to those defined under these Tenants. Set to ['all'] for all Tenants (default).
            # This list also limits Tenants included by `always_include_vrfs_in_tenants`.
            tenants: # default=['all']
              - <str>

            # Limit configured VLANs to those matching the given tags. Set to ['all'] for all VLANs (default).
            tags: # default=['all']
              - <str>

            # Limit configured Network Services to those defined under these VRFs. Set to ['all'] for all VRFs (default).
            # This list also limits VRFs included by `always_include_vrfs_in_tenants`.
            allow_vrfs: # default=['all']
              - <str>

            # Prevent configuration of Network Services defined under these VRFs.
            # This list prevents the given VRFs to be included by any other filtering mechanism.
            deny_vrfs: # default=['all']
              - <str>

            # List of tenants where VRFs will be configured even if VLANs are not included in tags.
            # Useful for L3 "border" leaf.
            always_include_vrfs_in_tenants:
              - <str>
            # Only configure VLANs, SVIs, VRFs in use by connected endpoints or downstream L2 switches.
            # Note! This feature only considers configuration managed by eos_designs.
            # This excludes structured_config, custom_structured_configuration_, raw_eos_cli, eos_cli, custom templates, configlets etc.
            only_vlans_in
```

## How to test

Molecule case with all combinations.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml [hunk 1]
```diff
@@ -507,14 +507,40 @@ $defs:
...
+                description: |-
+                  Limit configured Network Services to those defined under these VRFs. Set to ['all'] for all VRFs (default).
+                  This list also limits VRFs included by `always_include_vrfs_in_tenants`.
+                items:
+                  type: str
+                  convert_types:
+                    - int
+                default:
+                  - all
+              not_vrfs:
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml]: I would suggest changing key name to  `exclude_vrfs:` or `deny_vfrs:`

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml [hunk 2]
```diff
@@ -507,14 +507,40 @@ $defs:
...
                 default:
                   - all
               tags:
                 type: list
+                description: Limit configured VLANs to those matching the given tags. Set to ['all'] for all VLANs (default).
+                items:
+                  type: str
+                default:
+                  - all
+              vrfs:
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml]: I would suggest to change to the following key name:
```suggestion
              allow_vrfs:
```

**Review by carlbuchmann — APPROVED** [review]
- Review descriptions and keys, small suggestion better key name to make it clear.
- reviewed test case
- tested in lab

LGTM

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type.schema.yml]: Agreed this makes the data model much more intuitive. I have changed to `allow_vrfs` and `deny_vrfs`.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/host_vars/filter.vrfs.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+l3leaf:
+  nodes:
+    - name: filter.vrfs
+      id: 1
+      loopback_ipv4_pool: 10.0.0.0/24
+      bgp_as: 65001
+      vtep: false
+      filter:
+        deny_vrfs: [VRF1, VRF4]
+        allow_vrfs: [VRF1, VRF2, VRF4, VRF5]
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/host_vars/filter.vrfs.yml:13]: I understand why VRF1 is in both BUT probably would be good to state that it is to exercise the second part of the AND logic in our filtering function (the test is good just it is not clear why there is allow + deny ;)

**Review by gmuloc — APPROVED** [review]
LGTM - just adding one comment to explain the curiosity of the test in details would be good (that is why VRF are in both allow and deny list to really exercise the deny part of the python code)

**Review by gmuloc — CHANGES_REQUESTED** [review]
Please add comment on test

**Review by gmuloc — APPROVED** [review]
LGTM

**Pull request merged**